### ISSUE TITLE
Unexpected result from mlx.core.vjp

### ISSUE THREAD

**Issue by abdussamettrkr** [ml-explore/mlx#628]
**Describe the bug**
Obtained unexpected grad values from mlx.core.vjp.

**To Reproduce**

```python
import mlx.core as mx
def func(inp1, inp2):
    res = inp1 - inp2
    return res ** inp2
_, grad = mx.vjp(func,[mx.ones(2), mx.ones(2)],[mx.ones(2)*123])
print("Grad:", grad)
```
Prints: 
Grad: [array([123, 123], dtype=float32), array([nan, nan], dtype=float32)]

```python
import jax
def func(inp1, inp2):
    res = inp1 - inp2
    return res ** inp2

_, vjp_fn = jax.vjp(func, jax.numpy.ones(2), jax.numpy.ones(2))
grad =vjp_fn(jax.numpy.ones(2)*123)
print("Grad:", grad)
```
Prints:
Grad: (Array([123., 123.], dtype=float32), Array([-123., -123.], dtype=float32))

**Expected behavior**
Same result with Jax.

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.2.1
 - Version: 0.1.0

**Comment by awni** [comment]
Yea this is an edge case because $\frac{d}{dx} 0^x = 0^x \log 0 = 0 * -\infty$. That gives the `NaN`.. We could try and make it such that $0 \log 0 = 0$. That might require a `mx.where` which can handle `inf`.

**Comment by abdussamettrkr** [comment]
Yes. I think, obtaining consistent results with  Jax and Torch is better

**Comment by awni** [comment]
We can fix this once #598 is fixed.
